Modules/Module_Import/Import/xASL_imp_DetermineStructureFromSourcedata.m

```diff
@@ -240,42 +258,38 @@ function xASL_imp_PrintRun(x,thisSubject,thisVisit,thisRun)
     %% Preallocate space for (global) counts
     [x.importOverview.(sFieldName),x.importOverview.(sFieldName).(vFieldName)] = ...
         xASL_imp_PreallocateGlobalCounts(x.modules.import.nSubjects,x.importOverview.(sFieldName),x.importOverview.(sFieldName).(vFieldName));
-    
 
 end
 
-
+% -----------------------------------------------------------------
 %% Add fields of this subject/visit
+% -----------------------------------------------------------------
 function x = xASL_imp_thisSubjectVisit(x,sFieldName,vVisitIDs,vFieldName)
 
-
     % Visits (= sessions in BIDS)
     x = xASL_imp_AddVisitNames(x,sFieldName);
-    
-    
+   
     % Sessions (= run in BIDS)
     x = xASL_imp_AddSessionNames(x,sFieldName,vFieldName,vVisitIDs);
     
-    
     % Scans (= actual DICOM data)
     x = xASL_imp_AddScanNames(x,sFieldName,vFieldName,vVisitIDs);
-    
 
     % Sessions (= runs in BIDS)
     x = xASL_imp_AddSessions(x,sFieldName,vFieldName);
-
-
 end
 
-
+% -----------------------------------------------------------------
 %% Add visit names
+% -----------------------------------------------------------------
 function x = xASL_imp_AddVisitNames(x, sFieldName)
 
     if isempty(x.modules.import.imPar.visitNames)
         if isempty(x.importOverview.(sFieldName).visitIDs)
+			% In case that visit names are not filled, we name the visit by their number
             x.modules.import.imPar.visitNames = cell(x.importOverview.(sFieldName).nVisits,1);
             for iVisit=1:x.importOverview.(sFieldName).nVisits
-                x.modules.import.imPar.visitNames{iVisit} = sprintf('ASL_%g', iVisit);
+                x.modules.import.imPar.visitNames{iVisit} = xASL_num2str(iVisit);
```

Comment:
I coded xASL_num2str such that it can function for entire cell arrays at once without having to loop over all elements, but this may need testing. So you can ignore and resolve this if you want as well :)

Reply:
OK. But better to loop over cells ourselves.